src/stream/src/executor/temporal_join.rs

```diff
@@ -412,117 +586,96 @@ impl<K: HashKey, S: StateStore, const T: JoinTypePrimitive> TemporalJoinExecutor
-                                .filter_map(|(vis, key)| if vis { Some(key) } else { None });
-                            self.right_table
-                                .fetch_or_promote_keys(to_fetch_keys, epoch)
-                                .await?;
-                            for (r, key) in chunk.rows_with_holes().zip_eq_debug(keys.into_iter()) {
-                                let Some((op, left_row)) = r else {
-                                    continue;
-                                };
-                                if key.null_bitmap().is_subset(&null_matched)
-                                    && let join_entry = self.right_table.force_peek(&key)
-                                    && !join_entry.is_empty()
-                                {
-                                    for right_row in join_entry.cached.values() {
-                                        row_matched_bitmap_builder.append(true);
-                                        if let Some(chunk) =
-                                            builder.append_row(op, left_row.chain(right_row))
-                                        {
-                                            let row_matched =
-                                                std::mem::take(&mut row_matched_bitmap_builder)
-                                                    .finish();
-                                            yield (chunk, row_matched);
-                                        }
-                                    }
-                                } else if T == JoinType::LeftOuter {
-                                    row_matched_bitmap_builder.append(false);
-                                    if let Some(chunk) = builder.append_row(
-                                        op,
-                                        left_row.chain(risingwave_common::row::repeat_n(
-                                            DatumRef::None,
-                                            right_size,
-                                        )),
-                                    ) {
-                                        let row_matched =
-                                            std::mem::take(&mut row_matched_bitmap_builder)
-                                                .finish();
-                                        yield (chunk, row_matched);
-                                    }
-                                }
-                            }
-                            if let Some(chunk) = builder.take() {
-                                let row_matched =
-                                    std::mem::take(&mut row_matched_bitmap_builder).finish();
-                                yield (chunk, row_matched);
-                            }
+                    let epoch = prev_epoch.expect("Chunk data should come after some barrier.");
+
+                    let full_schema = full_schema.clone();
+
+                    if T == JoinType::Inner {
```

Comment:
It seems it didn't handle the case of inner join with other conditions.

Reply:
Good point, forget to handle that

Reply:
In fact, inner join with cond can be optimized to a normal `WHERE` clause. But I can handle it here first.